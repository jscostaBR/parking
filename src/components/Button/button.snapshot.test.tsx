import * as React from 'react'
import { Button } from '.'
import renderer from 'react-test-renderer'

describe('render Button and yours variations', () => {
  it('render correctly with type: primary and isValid: true', () => {
    const tree = renderer.create(<Button type='primary' isValid>Primary</Button>).toJSON();
    expect(tree).toMatchSnapshot()
  });
  it('render correctly with type: primary, isValid: false and disabled: true', () => {
    const tree = renderer.create(<Button type='primary' isValid={false}>Primary</Button>).toJSON();
    expect(tree).toMatchSnapshot()
  });
  it('render correctly with type: secondary, isValid: true', () => {
    const tree = renderer.create(<Button type='secondary' isValid>Secondary</Button>).toJSON();
    expect(tree).toMatchSnapshot()
  });
  it('render correctly with type: outlined, isValid: true', () => {
    const tree = renderer.create(<Button type='outlined' isValid>Outlined</Button>).toJSON();
    expect(tree).toMatchSnapshot()
  });
})
