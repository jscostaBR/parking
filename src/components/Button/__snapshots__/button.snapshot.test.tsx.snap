// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Button and yours variations render correctly with type: outlined, isValid: true 1`] = `
<button
  className="sc-bczRLJ sc-hKMtZM cEWaEo aWJSX"
>
  OUTLINED
</button>
`;

exports[`render Button and yours variations render correctly with type: primary and isValid: true 1`] = `
<button
  className="sc-bczRLJ sc-gsnTZi cEWaEo diHsI"
>
  PRIMARY
</button>
`;

exports[`render Button and yours variations render correctly with type: primary, isValid: false and disabled: true 1`] = `
<button
  className="sc-bczRLJ sc-gsnTZi cEWaEo diHsI"
>
  PRIMARY
</button>
`;

exports[`render Button and yours variations render correctly with type: secondary, isValid: true 1`] = `
<button
  className="sc-bczRLJ sc-dkzDqf cEWaEo eeEuWq"
>
  SECONDARY
</button>
`;
